haskell/lint.bzl

```diff
@@ -53,14 +59,31 @@ def _haskell_lint_aspect_impl(target, ctx):
         "--make",
     ])
 
-    args.add(expose_packages(
-        build_info,
-        lib_info,
-        use_direct = False,
-        use_my_pkg_id = None,
-        custom_package_caches = None,
-        version = ctx.rule.attr.version,
-    ))
+    args.add(lib_info.ghc_args if lib_info != None else bin_info.ghc_args)
+
+    # External libraries.
+    external_libs = set.from_list(build_info.external_libraries.values())
+    seen_libs = set.empty()
+    for lib in set.to_list(external_libs):
+        lib_name = get_lib_name(lib)
+        if not set.is_member(seen_libs, lib_name):
+            set.mutable_insert(seen_libs, lib_name)
+            if hs.toolchain.is_darwin:
+                args.add([
+                    "-optl-l{0}".format(lib_name),
+                    "-optl-L{0}".format(paths.dirname(lib.path)),
+                ])
+            else:
+                args.add([
+                    "-l{0}".format(lib_name),
+                    "-L{0}".format(paths.dirname(lib.path)),
+                ])
```

Comment:
Yeah, this should definitely be factored out (if it is necessary at all). I believe the code in `link.bzl` has changed in the meantime as well. Then it would be out of date before it was even merged …